analysers/commons/commons.py

```diff
@@ -0,0 +1,230 @@
+              "q3_min":q3Min, "q3_max":q3Max, "weighted_avg":weightedMean, \
+              "best": bestTrials, "worst": worstTrials, "average": averageTrials}
+    
+def computeMode(dataRDD):
+    from pyspark_cassandra import CassandraSparkContext
+    from pyspark_cassandra import RowFormat
+    from pyspark import SparkConf
+    
+    if dataRDD.isEmpty():
+        return (None, None)
+    
+    data = dataRDD.reduceByKey(lambda a, b: a + b) \
+            .map(lambda x: (x[1], x[0])) \
+            .sortByKey(0, 1) \
+            .collect()
+        
+    mode = list()
+    highestCount = data[0][0]        
+    for d in data:
+        if d[0] == highestCount:
+            mode.append(d[1])
+        else:
+            break
+    return (mode, highestCount)
+
+def computeModeMinMax(CassandraRDD, dataName):
+    if CassandraRDD.isEmpty():
+        return {"mode_min":None, "mode_max":None, \
+              "mode_min_freq":None, "mode_max_freq":None}
+    
+    modeMinValues = CassandraRDD.map(lambda x: (min(x[dataName+"_mode"]), x[dataName+"_mode_freq"])) \
+        .sortByKey(1, 1) \
+        .map(lambda x: (x[0], x[1])) \
+        .first()
+    modeMin = modeMinValues[0]
+    modeMinFreq = modeMinValues[1]
+    
+    modeMaxValues = CassandraRDD.map(lambda x: (max(x[dataName+"_mode"]), x[dataName+"_mode_freq"])) \
+        .sortByKey(0, 1) \
+        .map(lambda x: (x[0], x[1])) \
+        .first()
+    modeMax = modeMaxValues[0]
+    modeMaxFreq = modeMaxValues[1]
+    
+    return {"mode_min":modeMin, "mode_max":modeMax, \
+              "mode_min_freq":modeMinFreq, "mode_max_freq":modeMaxFreq}
+
+def cutNInitialProcesses(dataRDD, nToIgnore):
+    from pyspark_cassandra import CassandraSparkContext
+    from pyspark_cassandra import RowFormat
+    from pyspark import SparkConf
+    
+    if dataRDD.isEmpty():
+        return []
+    
+    processes = dataRDD.map(lambda r: r["process_definition_id"]) \
+            .distinct() \
+            .collect()
+    
+    maxTime = None
+    maxID = None
+    for p in processes:
+        time = dataRDD.filter(lambda r: r["process_definition_id"] == p) \
+            .map(lambda r: (r["start_time"], r["source_process_instance_id"])) \
+            .sortByKey(1, 1) \
+            .take(nToIgnore)
+        if len(time) < nToIgnore:
+            continue
+        else:
+            time = time[-1]
+        if maxTime is None or time[0] > maxTime:
+            maxTime = time[0]
+            defId = dataRDD.filter(lambda r: r["process_definition_id"] == p and r["start_time"] == maxTime) \
+                .map(lambda r: (r["source_process_instance_id"], 0)) \
+                .sortByKey(1, 1) \
+                .first()
+            maxID = defId[0]
+    
+    data = dataRDD.map(lambda r: (r["start_time"], r)) \
+            .sortByKey(1, 1) \
+            .map(lambda r: r[1]) \
+            .collect()
+    
+    index = -1
+    if maxID is not None:
+        for i in range(len(data)):
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Consider using enumerate instead of iterating with range and len](https://www.codacy.com/app/BenchFlow/analysers/file/2612114963/issues/source?bid=3301029&fileBranchId=3301030#l224)